R/pkg/inst/worker/worker.R

```diff
@@ -84,6 +84,10 @@ broadcastElap <- elapsedSecs()
 # as number of partitions to create.
 numPartitions <- SparkR:::readInt(inputCon)
 
+# If true, working for RDD
+# If false, working for DataFrame
+isDataFrame <- as.logical(SparkR:::readInt(inputCon))
```

Comment:
currently, SparkR is not a standalone package but an integral part of the Spark binary release. So it is assumed that the R worker script of correct matching version is always invoked. The protocol between JVM and the R worker is internal.